core/src/test/java/com/google/cloud/sql/core/SslSocketFactoryTest.java

```diff
@@ -170,20 +177,43 @@ public void create_throwsErrorForInvalidInstanceRegion() throws IOException {
         .thenReturn(
             new DatabaseInstance()
                 .setBackendType("SECOND_GEN")
-                .setIpAddresses(ImmutableList.of(new IpMapping().setIpAddress("127.0.0.1")))
+                .setIpAddresses(ImmutableList.of(new IpMapping().setIpAddress(PUBLIC_IP)))
                 .setServerCaCert(new SslCert().setCert(TestKeys.SERVER_CA_CERT))
                 .setRegion("beer"));
 
     SslSocketFactory sslSocketFactory =
         new SslSocketFactory(new Clock(), clientKeyPair, credential, adminApi, 3307);
     try {
-      sslSocketFactory.create("foo:bar:baz");
+      sslSocketFactory.create("foo:bar:baz", Arrays.asList("PRIMARY"));
       fail();
     } catch (IllegalArgumentException e) {
       assertThat(e.getMessage()).startsWith("Incorrect region value");
     }
   }
 
+  @Test
+  public void create_successfulPrivateConnection()
```

Comment:
Is there a specific reason for the underscore?

Reply:
The general style within Google for test naming is usually something like if there's a class Foo with a method bar, then in FooTest, if we want to test that bar works when baz is true, you would name the test method bar_bazIsTrue or something.